File: Introduction_to_Alogrithm/rod_cutting_recursive.cpp
# include <iostream>
# include <vector>
# incldue <algorithm>
using namespace std;
auto max_price(vector<int> &price,int len)
{
 if(len == 0)
   return 0;
 auto max_val = INT_MIN;
 for(int i=0;i<len;i++)
   max_val = max(max_val,price[i]+max_price(price,len-1-i));
return max_val;
}
int main()
{
vector<int> price_table = {1,5,8,9,10,17,17,20};
cout<<max_price(price_table,4);
}

File: Introduction_to_Algorithms/rod_cutting_recursive.rb
def max_price(price,len)
	return 0 if len.zero?
   max_val = -(Float::INFINITY)
   (0...len).each do |i|
    max_val = [max_val,price[i]+max_price(price,len-i-1)].max
   end
return max_val;
end
price = [1,5,8,9,10,17,17,20]
puts max_price(price,8)